headless/core/src/main/java/com/tencent/supersonic/headless/core/utils/ComponentFactory.java

```diff
@@ -74,7 +74,7 @@ public static <T> T getBean(String name, Class<T> tClass) {
     }
 
     private static void initQueryOptimizer() {
-        queryOptimizers.put("DetailQuery", getBean("DetailQuery", DetailQuery.class));
+        queryOptimizers.put("DetailQuery", getBean("DetailQuery", DetailQueryOptimizer.class));
```

Comment:
queryOptimizers.put("DetailQuery", getBean("DetailQuery",xxxx)

Perhaps here it needs to be changed to DetailQueryOptimizer.